plugins/heroku/src/herokuClient.ts

```diff
@@ -1,5 +1,30 @@
-import Heroku from 'heroku-client'
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { styles } from '@dotcom-tool-kit/logger'
+import { highlight } from 'cli-highlight'
+import Heroku, { HerokuError } from 'heroku-client'
 
 const HEROKU_AUTH_TOKEN = process.env.HEROKU_AUTH_TOKEN
 
-export default new Heroku({ token: HEROKU_AUTH_TOKEN })
+const client = new Heroku({ token: HEROKU_AUTH_TOKEN })
+
+// the Heroku client doesn't give particularly useful error messages, so we require an additional context
+// string to be included when calling the API which we include in custom error handling
+export async function herokuWithContext<T>(
+  context: string,
```

Comment:
**suggestion**: instead of passing in a string context with this call, could we extract the error handling into a separate function, to be used like <code>.catch(handleHerokuError(\`promoting app ID ${appId} to production`))</code>? would be cleaner than the callback wrapper

Reply:
the reason I went with this approach was to _force_ a context be applied when using the `heroku-client`. your alternative is definitely cleaner but I worry people will forget to use it if we add new Heroku API calls to the plugin. what do you think?

Reply:
that's a good point, but i think having the callback makes it a bit non-obvious what this function is actually doing when you're reading a call? i don't forsee us changing the `heroku` plugin much, and imo we should be more diligent about things like error messages in reviews anyway